"RuneLite Obsidian" Client Redesign

psikoi opened the pull request:
This redesign is based on issue #1342 and it attempts to take the "default" look out of runelite and hopefully improve it.

The game frame is now darker, and uses shades of grey to seperate elements instead of borders, it also contains new custom ui components, they are documented and can be used elsewhere on the project.
Every commit is (I think) properly detailed. So you'll find a lot more information there.

Progress:
- [x] Client recolour
- [x] G.E Offers
- [x] G.E Search
- [x] Hiscores
- [x] Exp trackers
- [x] Configs (slight modifications)
- [x] Info
- [x] Notes (slight modifications)
- [x] Kourend Library (slight modifications)
- [x] Feed (slight modifications)
- [x] Farming
- [x] New side bar icons


## Grand Exchange Offers:
![](https://i.gyazo.com/1ea8d2b796e7153d07c8116412782c13.png)
![](https://i.gyazo.com/aff267d949c9cb2204619d0b36a25107.png)





## Grand Exchange Search:
![](https://i.gyazo.com/2e8bea8a04f54afe23128a19df616553.png)
![](https://i.gyazo.com/adf19035e13b14d302606a5cb763830b.png)
![](https://i.gyazo.com/965390bfbf3a48acc47baf87d848d283.png)
![](https://i.gyazo.com/48f47c108c0ecdd71425f5abb77e8283.png)





## Hiscores:
![](https://i.gyazo.com/be96d4616a44d08855709189da00365b.png)
![](https://i.gyazo.com/124a022b59af37ad18373d415b1f55e1.png)



## Configs:
![](https://i.gyazo.com/d07c34a4eea058a7f97e5dc9dd572e7d.png)





## Kourend Library:
![](https://i.gyazo.com/06ef6c803983214abeb21b97b7164d77.png)





## Notes:
![](https://i.gyazo.com/1427b6dd3d63498b713e8938f2dfe583.png)





## News Feed:
![](https://i.gyazo.com/d276440c0681e98a9082ba0cb19e2fbd.png)




## Exp trackers:
![](https://i.gyazo.com/e1fabc481e09ed569aa7d362845b272a.png)
![](https://i.gyazo.com/54d53bb42fc2c4f5dcb3a6c11ea68a51.png)
![](https://i.gyazo.com/222c6464a51a0caa4d18479d6cfb53a7.png)




## Info:
![](https://i.gyazo.com/c20a5ed8805cce3a12e7a0159c94f21f.png)




## Farming:
![](https://i.gyazo.com/3bc97741677d7788a06f4a167ec363df.png)
![](https://i.gyazo.com/25054a87ca7067307b346788779c84ea.png)

deathbeam commented on runelite-client/src/main/java/net/runelite/client/ui/PluginPanel.java:
```diff
@@ -70,17 +69,20 @@ public PluginPanel()
 		wrappedPanel.add(scrollPane, BorderLayout.CENTER);
 	}
 
+	public PluginPanel(boolean disableScrolling) {
```
You can just do this directly by calling getScrollPane().setVerticalScrollBarPolicy(ScrollPaneConstants.VERTICAL_SCROLLBAR_NEVER);
instead of creating multiple constructors. Or even by adding protected method. New constructor is probably bad idea.

deathbeam commented on runelite-client/src/main/java/net/runelite/client/ui/ClientUI.java:
```diff
@@ -403,6 +405,7 @@ public void layoutContainer(Container parent)
 			frame.toFront();
 			requestFocus();
 			giveClientFocus();
+			UIManager.put(SubstanceSynapse.COLORIZATION_FACTOR, 1f);
```
Move this to SwingUtil.setupDefaults

deathbeam commented:
I believe this is not supposed to happen

![screenie](https://user-images.githubusercontent.com/5115805/38686836-94223f12-3e75-11e8-82bb-f593b1f1406d.png)

psikoi commented:
- Improved Resizable scrolling
- Moved colorization factor property to SwingUtil

psikoi commented:
Fixed item name labels for longer named items (Ex: Dragon scimitar ornament kit). These items would force the panel to introduce an horizontal scroll bar so I added a width limit.

psikoi commented:
Applied the new dark custom scroll bar theme to all of side plugin panels.

Before:
![](https://i.gyazo.com/b38c62f0a7a890b0fed23fc166fa06ea.png)

After:
![](https://i.gyazo.com/8bcd22a6f05ba4c2d55b96ca0dd37a66.png)

deathbeam commented on runelite-client/src/main/java/net/runelite/client/plugins/grandexchange/GrandExchangePanel.java:
```diff
@@ -54,11 +59,16 @@
 
+	private MaterialTabGroup tabGroup;
+	private MaterialTab offersTab;
+	private MaterialTab searchTab;
```
These should be final, and hmm, probably also inline variables, not instance ones.

deathbeam replied:
THis is still not marked as final

deathbeam commented on runelite-client/src/main/java/net/runelite/client/ui/ColorScheme.java:
```diff
@@ -0,0 +1,24 @@
+	/* The orange color used for the branding's accents */
+	public static final Color BRAND_ORANGE = new Color(220, 138, 0);
+
+
```
Too many empty lines

deathbeam commented on runelite-client/src/main/java/net/runelite/client/ui/CustomScrollBarUI.java:
```diff
@@ -0,0 +1,105 @@
+ */
+public class CustomScrollBarUI extends BasicScrollBarUI
+{
+	private final Color THUMB_BACKGROUND;
```
lowerCase or change this to constant, maybe even move it to ColorScheme

deathbeam commented on runelite-client/src/main/java/net/runelite/client/ui/materialtabs/MaterialTab.java:
```diff
@@ -0,0 +1,130 @@
+
+	private boolean selected;
+
+	private final Border SELECTED_BORDER = new CompoundBorder(
```
Either make this constant or use correct naming (lowerCase)

deathbeam commented on runelite-client/src/main/java/net/runelite/client/ui/materialtabs/MaterialTab.java:
```diff
@@ -0,0 +1,130 @@
+		BorderFactory.createMatteBorder(0, 0, 1, 0, ColorScheme.BRAND_ORANGE),
+		BorderFactory.createEmptyBorder(5, 10, 5, 10));
+
+	private final Border UNSELECTED_BORDER = BorderFactory
```
See my previous comment

deathbeam commented on runelite-client/src/main/java/net/runelite/client/ui/materialtabs/MaterialTab.java:
```diff
@@ -0,0 +1,130 @@
+		}
+
+
+		addMouseListener(new MouseListener()
```
Use MouseAdapter

deathbeam commented on runelite-client/src/main/java/net/runelite/client/ui/materialtabs/MaterialTab.java:
```diff
@@ -0,0 +1,130 @@
+	private MaterialTabGroup group;
+
+	/* The tab's associated content display */
+	private JComponent content;
```
Why are this 2 above not final?

deathbeam commented on runelite-client/src/main/java/net/runelite/client/ui/materialtabs/MaterialTabGroup.java:
```diff
@@ -0,0 +1,104 @@
+{
+
+	/* The panel on which the content tab's content will be displayed on. */
+	private JPanel display;
```
Final

deathbeam commented on runelite-client/src/main/java/net/runelite/client/ui/materialtabs/MaterialTabGroup.java:
```diff
@@ -0,0 +1,104 @@
+	/* The panel on which the content tab's content will be displayed on. */
+	private JPanel display;
+	/* A list of all the tabs contained in this group. */
+	private ArrayList<MaterialTab> tabs;
```
Final, and initialize it here not in constructor

TheeWiki commented:
https://i.imgur.com/3jOOOiB.png
Should have a word wrap on the item name

psikoi commented:
@deathbeam all your requests have been applied and both G.E searches and Hiscores are complete (to my knowledge), feel free to test and review.

LeviSchuck commented on runelite-client/src/main/java/net/runelite/client/ui/IconTextField.java:
```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright (c) 2017, Adam <Adam@sigterm.info>
```
Why is Adam's name on this one? You wrote it

psikoi replied:
He had written the previous version, I rewrote it and moved it to another folder.

LeviSchuck commented on runelite-client/src/main/java/net/runelite/client/plugins/grandexchange/GrandExchangeSearchPanel.java:
```diff
@@ -120,6 +143,37 @@ void priceLookup(String item)
+		this.parentSize = parentSize;
+
+		resultsWrapper.setPreferredSize(calculatePrefSize());
+
```
Unnecessary extra line between resultsWrapper calls?

LeviSchuck commented on runelite-client/src/main/java/net/runelite/client/plugins/grandexchange/GrandExchangeSearchPanel.java:
```diff
@@ -120,6 +143,37 @@ void priceLookup(String item)
+	{
+		int availableHeight = parentSize.height - 100;
+
+		if (ITEMS_LIST.isEmpty() || ITEMS_LIST.size() > (availableHeight / 63))
```
This 63 seems to be a magic constant.
You do say in the javadoc above that the list entry height of an item is 58 + 5 (which adds up to 63)

Could you make a static final int that is named for this?
Suppose: `ITEM_LIST_BLOCK_HEIGHT = 58 + 5`  with a comment explaining that the entry height is 58 and the margin is 5. Then use that constant here instead.

LeviSchuck commented on runelite-client/src/main/java/net/runelite/client/ui/materialtabs/MaterialTabGroup.java:
```diff
@@ -0,0 +1,103 @@
+ * JPanel "display" will display the content associated with that tab.
+ * <p>
+ * How to use these tabs:
+ * 1 - Create displays (JPanels) for each tab
```
You are mixing html and markdown here. This will all bunch up on the same line.
You'll have to do something lame like 
```
<ol>
  <li> Create displays ...</li>
  <li> Create an empty JPanel ...</li>
...
</ol>
```

sethtroll commented:
I still think the gray color (#4D4D4D) in the GE panels should be darker to match the rest of the overlays (#1E1E1E) or at least darker than they are now. Anyone have thoughts on it?

psikoi commented:
I tried the darker gray look for the GE panels and it wasn't feeling right, but tomorrow when I have some free time I can try and darken it and see how people feel about it

sethtroll commented:
I think it should be consistent

Saer-Dev commented:
Looks great, looking forward to seeing this implemented!

psikoi commented:
The code is now ready to be reviewed and merged :D (finally)

Kamielvf commented on runelite-client/src/main/java/net/runelite/client/plugins/grandexchange/GrandExchangeOfferSlot.java:
```diff
@@ -28,38 +28,57 @@
-	private final TitledBorder itemName = BorderFactory.createTitledBorder("Nothing");
-	private final JLabel offerState = new JLabel("Text so the label has height");
-	private final JProgressBar progressBar = new JProgressBar();
+	private final JLabel itemName = new JLabel("Fire rune");
```
Seems like you included some testing stuff

psikoi replied:
Oops.

Kamielvf commented:
Another issue I found was it breaks the alt+click item price lookup, because it cannot find the component. [Stacktrace](https://hastebin.com/tuyalaquge.rb)

These components have some sort of background
![image](https://user-images.githubusercontent.com/35824069/39390740-599a2690-4a99-11e8-8c97-7a0107a1a227.png)

Dropdown selected/hover colours change the text to dark and makes it hard to read
![image](https://user-images.githubusercontent.com/35824069/39390749-7c13a32c-4a99-11e8-8273-8bcff5b032be.png)

Hiscore textfield does no longer search for your own username on double click

jplsek commented on runelite-client/src/main/java/net/runelite/client/plugins/grandexchange/GrandExchangeOfferSlot.java:
```diff
@@ -72,123 +91,212 @@
+				return quantity / 1000 + "K";
+			}
 
+			return NumberFormat.getNumberInstance(Locale.US).format(quantity)
```
For 273-278, why not use the StackFormatter class?

jplsek commented on runelite-client/src/main/java/net/runelite/client/plugins/grandexchange/GrandExchangeSearchPanel.java:
```diff
@@ -179,44 +258,39 @@ private void priceLookup(boolean exactMatch)
+
+			// unfortunately, as I have not found a automatic way to make the scroll pane resize
+			// to properly wrap the content, it has to recalculate its size every search
+			resultsWrapper.setPreferredSize(calculatePrefSize());
```
The notes plugin may be doing something similar to what you are looking for. I think it sets a custom layout to the parent panel to get the scroll pane to update automatically.

jplsek commented on runelite-client/src/main/java/net/runelite/client/plugins/hiscore/HiscorePanel.java:
```diff
@@ -100,13 +93,10 @@
 	 * Real skills, ordered in the way they should be displayed in the panel.
 	 */
-	private static final Set<HiscoreSkill> SKILLS = new LinkedHashSet<>(Arrays.asList(
+	private static final ArrayList<HiscoreSkill> SKILLS = new ArrayList<>(Arrays.asList(
```
Any reason why the type is an ArrayList instead of a List? It should be kept as the interface. It should look like: `private static final List<HiscoreSkill> SKILLS = new ArrayList<>(Arrays.asList...`

jplsek commented on runelite-client/src/main/java/net/runelite/client/plugins/xptracker/XpInfoBox.java:
```diff
@@ -236,47 +201,55 @@ void update()
 
-		LookupOp op = new LookupOp(lookup, new RenderingHints(null));
-		return op.filter(image, null);
+			valueStr = NumberFormat.getNumberInstance(Locale.US).format(value)
```
Same as before. Why not use the StackFormatter? If there is something specific about not using, it may be worth adding to it or creating a similar class for skills.

jplsek commented on runelite-client/src/main/java/net/runelite/client/ui/components/ProgressBar.java:
```diff
@@ -0,0 +1,122 @@
+ */
+public class ProgressBar extends JPanel
+{
+	private int maximumValue;
```
Use `@Setter` instead of creating your own methods. Same for the `value` field.

jplsek commented on runelite-client/src/main/java/net/runelite/client/ui/components/ThinProgressBar.java:
```diff
@@ -0,0 +1,76 @@
+ */
+public class ThinProgressBar extends JPanel
+{
+	private int maximumValue;
```
Same as the previous review.

jplsek commented on runelite-client/src/main/java/net/runelite/client/ui/components/ThinProgressBar.java:
```diff
@@ -0,0 +1,76 @@
+
+	public double getPercentage()
+	{
+		if (value == 0)
```
Why is this if statement necessary? `(0 * 100) / x` is still 0.

jplsek replied:
That's a weird test then. That means no test that accounts for `value = 1`, and `maximumValue = 0`, which would throw an error. I'm guessing there is no test because there is no way for `value` to be greater than zero if the `maximumValue` is zero... Either way, it would be better if the check was `if (maximumValue == 0)` since that would make more sense imo.

jplsek commented on runelite-client/src/main/java/net/runelite/client/ui/components/materialtabs/MaterialTab.java:
```diff
@@ -0,0 +1,106 @@
+	/* The tab's associated content display */
+	private final JComponent content;
+
+	private boolean selected;
```
What's the use for the `selected` field? It's never assigned.

jplsek commented on runelite-client/src/main/java/net/runelite/client/plugins/info/InfoPanel.java:
```diff
@@ -105,51 +130,102 @@ void init()
+	 * Builds a link panel with a given icon, text and url to redirect to.
+	 */
+	private JPanel buildLinkPanel(String iconName, String topText, String bottomText, String url)
+	{
```
There are these extra "new lines" in multiple files, or after a brace. IMO, there shouldn't need to be more than 1 blank line in places, and there shouldn't be a blank line after a brace. That's a reason to have braces on a new line (it already adds whitespace to the code). Those should probably be removed.

As an example for multiple new lines, in this file, line 132, 133 and 150, 151 can just be one line.

MagicfTail commented:
Double-clicking the search bar in the HiScore plugin doesn't search for your own name.

psikoi commented:
@MagicfTail it does if you click the search icon, i'll extend it to the text field.

MagicfTail commented:
Tbh, i like that a lot better

psikoi commented:
The icon click? me too, but it's not really my call.

PariksitDogra commented:
Hey psikoi, do you have any ETA on the obsidian mode?

psikoi commented:
@PariksitDogra Not really, the farming plugin just came out so I'm working on restyling that, then after that it's a matter of making sure everyone's happy with the code and get it hopefully approved.

jplsek commented on runelite-client/src/main/java/net/runelite/client/ui/components/PluginErrorPanel.java:
```diff
@@ -0,0 +1,53 @@
+package net.runelite.client.ui.components;
```
Add license.

psikoi replied:
Completely forgot about those :s thanks man

jplsek commented on runelite-client/src/main/java/net/runelite/client/ui/ColorScheme.java:
```diff
@@ -0,0 +1,48 @@
+package net.runelite.client.ui;
```
Add license.

jplsek commented on runelite-client/src/main/java/net/runelite/client/ui/components/ComboBoxListRenderer.java:
```diff
@@ -0,0 +1,41 @@
+package net.runelite.client.ui.components;
```
Add license.

jplsek commented on runelite-client/src/main/java/net/runelite/client/ui/components/ThinProgressBar.java:
```diff
@@ -0,0 +1,70 @@
+package net.runelite.client.ui.components;
```
Add license.

deathbeam commented on runelite-client/src/main/java/net/runelite/client/ui/PluginToolbar.java:
```diff
@@ -39,7 +39,7 @@
 {
 	private final EventBus eventBus;
-	private final TreeSet<NavigationButton> buttons = new TreeSet<>(Comparator.comparing(NavigationButton::getName));
+	private final TreeSet<NavigationButton> buttons = new TreeSet<>(Comparator.comparing(NavigationButton::getPriority));
```
You should order the buttons both by name and priority

psikoi commented:
@deathbeam added priority + name comparator, thanks.

Abextm commented on runelite-client/src/main/java/net/runelite/client/plugins/info/InfoPanel.java:
```diff
@@ -105,51 +130,98 @@ void init()
+	private JPanel buildLinkPanel(String iconName, String topText, String bottomText, String url)
+	{
+		ImageIcon icon;
+		try
```
Most of this should be in a static block

psikoi commented on runelite-client/src/main/java/net/runelite/client/ui/skin/ObsidianSkin.java:
```diff
@@ -0,0 +1,219 @@
+/*
+ * Copyright (c) 2018, Tomas Slusny <slusnucky@gmail.com>
```
deathbeam first started the skinning, i just forgot to add my name to it I guess.

psikoi commented on runelite-client/src/main/java/net/runelite/client/ui/components/IconTextField.java:
```diff
@@ -0,0 +1,147 @@
+	public void lockInput()
+	{
+		this.blocked = true;
+		super.setBackground(backgroundColor);
```
Because it would be the same color as the locked background, it just changes the background color on the locking to prevent any hover effects

psikoi commented on runelite-client/src/main/java/net/runelite/client/ui/components/materialtabs/MaterialTabGroup.java:
```diff
@@ -0,0 +1,105 @@
+	/* The panel on which the content tab's content will be displayed on. */
+	private final JPanel display;
+	/* A list of all the tabs contained in this group. */
+	private final ArrayList<MaterialTab> tabs = new ArrayList<>();
```
Then i couldn't select using #get(index), correct?

Adam- replied:
This is probably small enough that a list doesn't matter. But it should be `List` not `ArrayList`

psikoi commented on runelite-client/src/main/java/net/runelite/client/plugins/grandexchange/GrandExchangeSearchPanel.java:
```diff
@@ -118,6 +157,35 @@ void priceLookup(String item)
+	 * If all items fit on screen, the size is set to their heights (58 height + 5 margin) combined,
+	 * if not, the size will be set to the full panel height - 100 pixels (enough room for the search bar and tabs).
+	 */
+	private Dimension calculatePrefSize()
```
There has to be, I just fiddled with it for hours and was getting frustrated so I left it as is until I figured out a better way. I'll work on it today.

psikoi commented on runelite-client/src/main/java/net/runelite/client/util/StackFormatter.java:
```diff
@@ -128,6 +129,62 @@ else if (quantity < 10_000_000)
+			{
+				return quantity / 1000 + "K";
+			}
+			return NumberFormat.getInstance(Locale.UK).format(quantity).substring(0, Integer.toString(quantity).length() - 1) + "K";
```
NUMBER_FORMATTER has no specific locale, so it wouldn't pass tests for me and Travis at the same time for example, as our locales are different.

deathbeam replied:
I would force the Locale of NUMBER_FORMATTER then, would solve a lot of issues we have

psikoi replied:
Okay

psikoi commented:
Updated preview images.

psikoi commented:
Updated the code to apply some of @Abextm's review suggestions:

- Added CustomScrollBarUI to the swing ui defaults (UIManager)
- Added copyright that was missing from some files
- Changed lockInput() and unlockInput() into setEditable(boolean) in IconTextField
- Forced US Locale in the StackFormatter class
- Changed MouseEvent to mouseClicked in HiscorePanel
- Added new method to SwingUtil that darkens/brightens an image and used that for click feedback effects on the Kourend and New Feeds plugins' refresh buttons (previously had a darker image for each)
- Moved image loading to a static block on the InfoPanel
- Other small code improvements (adding final to few variables, mering try/catch blocks, etc)

psikoi commented on runelite-client/src/main/resources/net/runelite/client/ui/skin/RuneLite.colorschemes:
```diff
@@ -0,0 +1,142 @@
+RuneLite Enabled {
```
I don't think it does, atleast not the /* */ format, i tried.

Kamielvf commented:
These components still have a different coloured background assigned to them:
Between these game size areas: (the `x` part)
![image](https://user-images.githubusercontent.com/35824069/39657586-89ba12b6-5009-11e8-8a2f-d3b4a9f3093f.png)
Dev tools:
![image](https://user-images.githubusercontent.com/35824069/39657594-9a6a6b38-5009-11e8-93cd-c07578cda49f.png)

psikoi commented:
- Fixed dev tools background color mismatch
- Fixed dimensions config item background color mismatch

Thanks @Kamielvf

jplsek commented:
There is a regression where the GE Search removes focus after pressing enter. (In the upstream version, I can press enter and continue to type, in this version it removes focus from the input)

psikoi commented:
@jplsek It was done on purpose to bring the ui back to it's default state, do you think it makes the user experience worse?

jplsek commented:
With the GE search, part of the problem is that the search isn't "fuzzy" enough. For example, I search "bron" and I get no results, so I have to click the search bar again to fully search "bronze".

However, since it was done on purpose, then it's probably fine.

Another issue I noticed is when selecting all characters in the search and the mouse is not hovering the input, the text background is the same as the input background. The text color is different, but the text background should also be different to be consistent. Example:
![ss 2018-05-04 21 27 26](https://user-images.githubusercontent.com/2388657/39658412-ffcca676-4fe1-11e8-9169-881063f0f522.png)

psikoi commented:
- Added new text selection colors (ty @jplsek)

![](https://i.gyazo.com/e7b887c3cfd42ca4982314d6aec2b490.png)

psikoi commented:
- Improved Grand Exchange Search scrolling behaviour. Thanks to @jplsek's patch and @Abextm and @Kamielvf help.

ON OTHER NEWS, 100 COMMENTS, DAMN.

deathbeam commented on runelite-client/src/main/java/net/runelite/client/ui/components/IconTextField.java:
```diff
@@ -0,0 +1,152 @@
+	private JTextField textField;
+
+	//to support gifs, the icon needs to be wrapped in a JLabel
+	private JLabel iconWrapperLabel;
```
Mark this field final

deathbeam commented on runelite-client/src/main/java/net/runelite/client/ui/components/IconTextField.java:
```diff
@@ -0,0 +1,152 @@
+ */
+public class IconTextField extends JPanel
+{
+	private JTextField textField;
```
Mark this field final

deathbeam commented on runelite-client/src/main/java/net/runelite/client/ui/components/PluginErrorPanel.java:
```diff
@@ -0,0 +1,78 @@
+{
+
+	private JLabel noResultsTitle = new JShadowedLabel();
+	private JLabel noResultsDescription = new JShadowedLabel();
```
Mark these 2 fields final

deathbeam commented on runelite-client/src/main/java/net/runelite/client/ui/components/ThinProgressBar.java:
```diff
@@ -23,54 +23,73 @@
+	@Setter
+	private int value;
+
+	private JPanel topBar;
```
Mark this field final

deathbeam commented on runelite-client/src/main/java/net/runelite/client/ui/components/ThinProgressBar.java:
```diff
@@ -23,54 +23,73 @@
+	public void setForeground(Color color)
 	{
-		setBorder(border);
+		if (topBar == null)
```
This nullcheck is not neccessary

psikoi replied:
It seems to be, I could't quite understand why, but it seems the setForeground gets called before the topBar is initialized, do you have any idea how this could be?

Abextm replied:
Its probably called in the superclass's constructor somewhere.

psikoi replied:
I could be reading it wrong, but I'm sure it only calls that method after the instance is inicialized.

deathbeam commented on runelite-client/src/main/java/net/runelite/client/ui/components/materialtabs/MaterialTab.java:
```diff
@@ -0,0 +1,110 @@
+	@Getter
+	private boolean selected;
+
+	private static final Border SELECTED_BORDER = new CompoundBorder(
```
Move static fields above instace fields (at start of the class)

deathbeam commented on runelite-client/src/main/java/net/runelite/client/ui/components/materialtabs/MaterialTab.java:
```diff
@@ -0,0 +1,110 @@
+	private final MaterialTabGroup group;
+
+	/* The tab's associated content display */
+	private final JComponent content;
```
Use @Getter annotation instead of method

deathbeam commented on runelite-client/src/main/java/net/runelite/client/plugins/grandexchange/GrandExchangeOfferSlot.java:
```diff
@@ -28,38 +28,55 @@
+	private static final String FACE_CARD = "FACE_CARD";
+	private static final String DETAILS_CARD = "DETAILS_CARD";
 
+	private JPanel container = new JPanel();
```
Mark this final

deathbeam commented on runelite-client/src/main/java/net/runelite/client/plugins/grandexchange/GrandExchangeOfferSlot.java:
```diff
@@ -28,38 +28,55 @@
+
+	private final ThinProgressBar progressBar = new ThinProgressBar();
+
+	private ImageIcon rightArrowIcon;
```
Move initialization of both icons to static { } initializer and make them constants

deathbeam commented on runelite-client/src/main/java/net/runelite/client/plugins/grandexchange/GrandExchangeOfferSlot.java:
```diff
@@ -72,123 +89,189 @@
+		faceCard.add(switchFaceViewIcon, BorderLayout.EAST);
+
+
+		/* -------------------------------------------------------------------------------------- */
```
So many new lines :O

psikoi replied:
My bad, was supposed to clean up afterwards, that was just a little visual seperator to keep me organized.

deathbeam commented on runelite-client/src/main/java/net/runelite/client/plugins/grandexchange/GrandExchangeSearchPanel.java:
```diff
@@ -36,35 +39,64 @@
-	private ImageIcon search;
+	private ItemClient itemClient;
+
+	private ImageIcon searchIcon;
```
Initialize this in static { } initializer and make these icons constants

deathbeam commented on runelite-client/src/main/java/net/runelite/client/plugins/grandexchange/GrandExchangeSearchPanel.java:
```diff
@@ -36,35 +39,64 @@
 	private final ScheduledExecutorService executor;
 
-	private ImageIcon search;
+	private ItemClient itemClient;
```
Mark this final and initialize it here, preferrably make this constant

psikoi replied:
What variable are you referring to?

deathbeam replied:
ItemClient

deathbeam commented on runelite-client/src/main/java/net/runelite/client/util/StackFormatter.java:
```diff
@@ -48,7 +49,7 @@
 	 * A number formatter
 	 */
-	private static final NumberFormat NUMBER_FORMATTER = NumberFormat.getInstance();
+	private static final NumberFormat NUMBER_FORMATTER = NumberFormat.getInstance(Locale.US);
```
Use UK Locale not US locale here

deathbeam commented on runelite-client/src/main/java/net/runelite/client/plugins/hiscore/HiscorePanel.java:
```diff
@@ -159,28 +154,30 @@ public HiscorePanel(HiscoreConfig config)
+				loadingIcon = new ImageIcon(this.getClass().getResource("loading_spinner.gif"));
 			}
-			search = new ImageIcon(icon);
+			searchIcon = new ImageIcon(searchImg);
```
static {} initializer, constants

deathbeam commented on runelite-client/src/main/java/net/runelite/client/plugins/hiscore/HiscorePanel.java:
```diff
@@ -306,171 +226,103 @@ public void mouseClicked(MouseEvent e)
+							return;
+						}
+
+						executor.execute(new Runnable()
```
executor.execute(HiscorePanel.this::lookup)

psikoi replied:
ah! I'm not familiar with that syntax, thanks.

deathbeam commented on runelite-client/src/main/java/net/runelite/client/plugins/kourendlibrary/KourendLibraryPanel.java:
```diff
@@ -53,7 +66,10 @@ void init()
 		GroupLayout layout = new GroupLayout(this);
 		setLayout(layout);
 
+		setBackground(ColorScheme.DARK_GRAY_COLOR);
```
I am starting to feel like that all these `setBackground(ColorScheme.DARK_GRAY_COLOR)` should be instead in PluginPanel

psikoi replied:
They are, however setOpaque seems to not work for some jpanels, leaving them with a different background than the PluginPanel, but for that example in specific, I think you're right.

deathbeam commented on runelite-client/src/main/java/net/runelite/client/plugins/info/InfoPanel.java:
```diff
@@ -52,6 +63,14 @@
+	private final JLabel loggedLabel = new JLabel();
+	private final JRichTextPane emailLabel = new JRichTextPane();
+
+	private static ImageIcon arrowRight;
```
UPPER_CASE all these icons

deathbeam commented on runelite-client/src/main/java/net/runelite/client/plugins/hiscore/HiscorePlugin.java:
```diff
@@ -95,12 +95,13 @@ protected void startUp() throws Exception
 		synchronized (ImageIO.class)
 		{
-			icon = ImageIO.read(getClass().getResourceAsStream("hiscore.gif"));
+			icon = ImageIO.read(getClass().getResourceAsStream("normal.png"));
```
Why was this changed from hiscore.gif to normal.png?

psikoi replied:
the straight-from-the-sprites hiscore.gif has a weird empty right side border, this makes it not horizontally centered.

deathbeam commented:
What about 120? 😄

devLotto commented:
What do you think of making it so it looks like the sidebar's margin is the same on both sides? I think it would look better.

Before: https://camo.githubusercontent.com/41fd6048f7af7dfcd6ff27120b93c9533fcce222/68747470733a2f2f692e6779617a6f2e636f6d2f35323962653031623761666334333961313662623737656335646362363564652e706e67

After: https://i.imgur.com/H59zKHY.png

I think the culprit is the 4px border around the whole client frame, so I guess a solution/hack could be to remove the right-side margin and make the left-side one 4px? Apparently actually removing the border just on the right side makes it so you cannot resize the frame from that side, which is obviously not desirable.

deathbeam commented:
That sounds like it would look great, but, with custom chrome disabled the 4px padding needs to be reverted back.

psikoi commented:
@devLotto I fiddled with this a bit and even asked on discord if there was a way to check if the custom chrome was enabled, so that I could remove the border, i was told there wasn't a way to check.

Saer-Dev commented:
Perhaps remove Custom Chrome or the Native Windows header altogether?  Don't think it would be as much of a controversial move if there is the Obsidian Theme available.  Perhaps if there was a white theme released as well this would be even less of an issue.

deathbeam commented:
I actually have #1998 that wants to remove the custom window chrome completely. But I think I would not worry much about neither in this PR and the easiest way is I guess to just leave the 4px there as it is now in this PR.

psikoi commented:
Updated the code to apply some of @deathbeam's review suggestions:

Most of these changes were small code improvement tweaks and clean-up, here's a few relevant ones:
- Quite a few of final modifiers were added to variables
- A few image loading blocks were moved to a static block, and the image variables were renamed to upper case
- NUMBER_FORMATTER's locale was changed to UK (StackFormatter)

Adam- commented on runelite-client/src/main/java/net/runelite/client/plugins/grandexchange/GrandExchangeSearchPanel.java:
```diff
@@ -26,46 +27,92 @@
 	private final ScheduledExecutorService executor;
 
-	private Icon search;
+	private final ItemClient itemClient = new ItemClient();
```
what is this for?

psikoi replied:
ItemClient? I'm not sure, I wasn't the one who implemented the fetching.

Adam- commented on runelite-client/src/main/java/net/runelite/client/plugins/hiscore/HiscorePanel.java:
```diff
@@ -101,13 +93,10 @@
 	 * Real skills, ordered in the way they should be displayed in the panel.
 	 */
-	private static final Set<HiscoreSkill> SKILLS = new LinkedHashSet<>(Arrays.asList(
+	private static final List<HiscoreSkill> SKILLS = new ArrayList<>(Arrays.asList(
```
This should just `Arrays.asList` or `ImmutableList.of`

Adam- commented on runelite-client/src/main/java/net/runelite/client/plugins/hiscore/HiscorePanel.java:
```diff
@@ -131,24 +120,47 @@
 
-	private List<JToggleButton> endpointButtons;
+	/* A list of all the selectable endpoints (ironman, deadman, etc) */
+	private ArrayList<JPanel> endPoints;
```
this should be `List`

Adam- commented on runelite-client/src/main/java/net/runelite/client/plugins/hiscore/HiscorePanel.java:
```diff
@@ -545,51 +378,84 @@ private void lookup()
 		}
 
+		/* Runescape usernames can't be longer than 12 characters long */
+		if (lookup.length() > 12)
```
this should be a constant somewhere

Adam- commented on runelite-client/src/main/java/net/runelite/client/plugins/info/InfoPanel.java:
```diff
@@ -105,51 +149,74 @@ void init()
+		actionsContainer.setOpaque(false);
+		actionsContainer.setLayout(new GridLayout(3, 1, 0, 10));
+
+		actionsContainer.add(buildLinkPanel(GITHUB_ICON, "Report an issue or", "make a suggestion", "https://github.com/runelite"));
```
These links should come from `runelite.properties`

psikoi commented:
Applied @Adam-'s review suggesions:
- Changed MaterialTabGroup's tabs to List (previously ArrayList)
- Removed useless variable itemClient in the Grand Exchange Search Panel
- Changed the skill list to immutable list on the hiscores panel
- Changed endpoints to List (previously ArrayList) on the hiscores panel
- Added constant variable to hold the maximum username length on the hiscores panel
- Added two new variables and getter methods to the runelite properties file containing a github link and patreon link
- Used the two new variables in the properties file on the info panel plugin.

deathbeam commented on runelite-client/src/main/java/net/runelite/client/plugins/grandexchange/GrandExchangeSearchPanel.java:
```diff
@@ -26,46 +27,89 @@
+	private final IconTextField searchBox = new IconTextField();
+
+	private static final GridBagConstraints CONSTRAINTS = new GridBagConstraints();
+	private static final CardLayout CARD_LAYOUT = new CardLayout();
```
Why are these 2 fields constants when you are modifying them in instance

deathbeam commented on runelite-client/src/main/java/net/runelite/client/plugins/grandexchange/GrandExchangeSearchPanel.java:
```diff
@@ -26,46 +27,89 @@
+	private static ImageIcon LOADING_ICON;
+	private static ImageIcon ERROR_ICON;
+
+	private static final List<GrandExchangeItems> ITEMS_LIST = new ArrayList<>();
```
Why is this constant

psikoi replied:
I don't know, I didn't write it.

deathbeam commented on runelite-client/src/main/java/net/runelite/client/plugins/kourendlibrary/KourendLibraryPanel.java:
```diff
@@ -70,21 +86,65 @@ void init()
+			BufferedImage resetClickImg;
+			synchronized (ImageIO.class)
+			{
+				resetImg = ImageIO.read(KourendLibraryPanel.class.getResourceAsStream("reset.png"));
```
Probably make these 2 images constants too

deathbeam commented on runelite-client/src/main/java/net/runelite/client/plugins/xptracker/SkillColor.java:
```diff
@@ -0,0 +1,66 @@
+	HUNTER(92, 89, 65),
+	CONSTRUCTION(130, 116, 95);
+
+	private final Color color;
```
Use getter annotation

deathbeam commented on runelite-client/src/main/java/net/runelite/client/plugins/xptracker/XpInfoBox.java:
```diff
@@ -221,46 +187,47 @@ private void rebuildAsync(boolean updated, XpSnapshotSingle xpSnapshotSingle)
 
-		LookupOp op = new LookupOp(lookup, new RenderingHints(null));
-		return op.filter(image, null);
+		return "<html><body style = 'color:#a5a5a5'>" + key + "<span style = 'color:white'>" + valueStr + "</span></body></html>";
```
Again the magical a5a5a5

deathbeam commented on runelite-client/src/main/java/net/runelite/client/plugins/xptracker/XpPanel.java:
```diff
@@ -96,14 +126,18 @@
+		errorPanel.setContent("Exp trackers", "You have not gained experience yet.");
+		errorPanel.setBorder(new EmptyBorder(0, 0, 0, 0));
+		add(errorPanel);
+
```
Newline

deathbeam commented on runelite-client/src/main/java/net/runelite/client/plugins/xptracker/XpPanel.java:
```diff
@@ -150,30 +184,26 @@ void updateSkillExperience(boolean updated, Skill skill, XpSnapshotSingle xpSnap
-		else
+
+		SwingUtilities.invokeLater(() ->
 		{
```
Remove the unnecessary braces

deathbeam commented on runelite-client/src/main/java/net/runelite/client/ui/components/PluginErrorPanel.java:
```diff
@@ -0,0 +1,78 @@
+		setVisible(false);
+	}
+
+
```
Newline

deathbeam commented on runelite-client/src/main/java/net/runelite/client/ui/components/PluginErrorPanel.java:
```diff
@@ -0,0 +1,78 @@
+		noResultsDescription.setText("<html><body style = 'text-align:center'>" + description + "</body></html>");
+		setVisible(true);
+	}
+
```
Newline

deathbeam commented on runelite-client/src/main/java/net/runelite/client/ui/components/PluginErrorPanel.java:
```diff
@@ -0,0 +1,78 @@
+ */
+public class PluginErrorPanel extends JPanel
+{
+
```
Newline

deathbeam commented on runelite-client/src/main/java/net/runelite/client/ui/components/ProgressBar.java:
```diff
@@ -0,0 +1,116 @@
+
+	private JLabel leftLabel = new JShadowedLabel();
+	private JLabel rightLabel = new JShadowedLabel();
+	private JLabel centerLabel = new JShadowedLabel();
```
Final to all 3

deathbeam commented on runelite-client/src/main/java/net/runelite/client/ui/components/materialtabs/MaterialTab.java:
```diff
@@ -0,0 +1,114 @@
+		setForeground(Color.GRAY);
+		selected = false;
+	}
+
```
Newline

deathbeam commented on runelite-client/src/main/java/net/runelite/client/ui/components/materialtabs/MaterialTabGroup.java:
```diff
@@ -0,0 +1,118 @@
+ */
+public class MaterialTabGroup extends JPanel
+{
+
```
Newline

deathbeam commented on runelite-client/src/main/java/net/runelite/client/ui/components/materialtabs/MaterialTabGroup.java:
```diff
@@ -0,0 +1,118 @@
+	/* Returns the tab on a certain index. */
+	public MaterialTab getTab(int index)
+	{
+
```
Newline

deathbeam commented on runelite-client/src/main/java/net/runelite/client/ui/components/materialtabs/MaterialTabGroup.java:
```diff
@@ -0,0 +1,118 @@
+		}
+
+		return true;
+
```
Newline

deathbeam commented on runelite-client/src/main/java/net/runelite/client/ui/components/materialtabs/MaterialTabGroup.java:
```diff
@@ -0,0 +1,118 @@
+		return true;
+
+	}
+
```
Newline

deathbeam commented on runelite-client/src/main/java/net/runelite/client/util/StackFormatter.java:
```diff
@@ -128,6 +129,62 @@ else if (quantity < 10_000_000)
+			}
+			return NUMBER_FORMATTER.format(quantity).substring(0, Integer.toString(quantity).length() - 7) + "B";
+		}
+
```
Newline

deathbeam commented on runelite-client/src/main/java/net/runelite/client/util/SwingUtil.java:
```diff
@@ -77,17 +82,57 @@ public static void setupDefaults()
 		// Fixes black boxes under popups that are above the game applet.
 		System.setProperty("jgoodies.popupDropShadowEnabled", "false");
 
+		UIManager.put("ScrollBarUI", CustomScrollBarUI.class.getName());
```
Move the custom UI modifications together

deathbeam commented on runelite-client/src/main/java/net/runelite/client/RuneLiteProperties.java:
```diff
@@ -82,4 +84,17 @@ public String getDiscordInvite()
+	{
+		return properties.getProperty(PATREON_LINK);
+	}
+
```
Newline

deathbeam commented on runelite-client/src/main/java/net/runelite/client/RuneLiteProperties.java:
```diff
@@ -82,4 +84,17 @@ public String getDiscordInvite()
+
 }
+
+
```
NewlineS

deathbeam commented on runelite-client/src/main/java/net/runelite/client/plugins/config/ConfigPanel.java:
```diff
@@ -150,42 +163,59 @@ final void rebuildPluginList()
+				newChildren.put(pluginName, groupPanel);
+			});
 
 
```
Newline

deathbeam commented on runelite-client/src/main/java/net/runelite/client/plugins/farmingtracker/FarmingTrackerPanel.java:
```diff
@@ -77,63 +89,108 @@
+				if (patch.getImplementation() != lastImpl && !Strings.isNullOrEmpty(patch.getImplementation().getName()))
+				{
+					JLabel groupLabel = new JLabel(patch.getImplementation().getName());
+					if (first)
```
Add newline above

deathbeam commented on runelite-client/src/main/java/net/runelite/client/plugins/farmingtracker/FarmingTrackerPanel.java:
```diff
@@ -77,63 +89,108 @@
+					{
+						groupLabel.setBorder(new EmptyBorder(15, 0, 0, 0));
+					}
+					groupLabel.setFont(FontManager.getRunescapeSmallFont());
```
Add newline above

deathbeam commented on runelite-client/src/main/java/net/runelite/client/plugins/grandexchange/GrandExchangeOfferSlot.java:
```diff
@@ -28,38 +28,71 @@
+	private final ThinProgressBar progressBar = new ThinProgressBar();
+
+	private static ImageIcon RIGHT_ARROW_ICON;
+	private static ImageIcon LEFT_ARROW_ICON;
```
Make these both final, and move them next to other constants

deathbeam commented on runelite-client/src/main/java/net/runelite/client/plugins/grandexchange/GrandExchangePanel.java:
```diff
@@ -26,58 +27,135 @@
 	{
+		if (newOffer == null || newOffer.getState() == GrandExchangeOfferState.EMPTY)
+		{
+
```
Newline

deathbeam commented on runelite-client/src/main/java/net/runelite/client/plugins/grandexchange/GrandExchangeSearchPanel.java:
```diff
@@ -26,46 +27,89 @@
-	private Icon search;
+	private static ImageIcon SEARCH_ICON;
+	private static ImageIcon LOADING_ICON;
+	private static ImageIcon ERROR_ICON;
```
Make these 3 fields final and move them at beginning of the class

deathbeam commented on runelite-client/src/main/java/net/runelite/client/plugins/grandexchange/GrandExchangeSearchPanel.java:
```diff
@@ -26,46 +27,89 @@
+	private final PluginErrorPanel errorPanel = new PluginErrorPanel();
+
+	private static final String ERROR_PANEL = "ERROR_PANEL";
+	private static final String RESULTS_PANEL = "RESULTS_PANEL";
```
Move constants at beginning of the class

deathbeam commented on runelite-client/src/main/java/net/runelite/client/plugins/grandexchange/GrandExchangeSearchPanel.java:
```diff
@@ -26,46 +27,89 @@
+	private JScrollPane resultsWrapper;
+
+	/*  The error panel, this displays an error message */
+	private final PluginErrorPanel errorPanel = new PluginErrorPanel();
```
Move final fields together

deathbeam commented on runelite-client/src/main/java/net/runelite/client/plugins/hiscore/HiscorePanel.java:
```diff
@@ -131,24 +123,47 @@
 
-	private List<JToggleButton> endpointButtons;
+	/* A list of all the selectable endpoints (ironman, deadman, etc) */
+	private List<JPanel> endPoints = new ArrayList<>();
```
mark final

deathbeam commented on runelite-client/src/main/java/net/runelite/client/plugins/hiscore/HiscorePanel.java:
```diff
@@ -131,24 +123,47 @@
 
+	private static ImageIcon SEARCH_ICON;
+	private static ImageIcon LOADING_ICON;
+	private static ImageIcon ERROR_ICON;
```
Mark these 3 fields final, and move the constants together at top of the class

deathbeam commented on runelite-client/src/main/java/net/runelite/client/plugins/hiscore/HiscorePanel.java:
```diff
@@ -306,172 +226,94 @@ public void mouseClicked(MouseEvent e)
+
+				endPoints.add(panel);
+				endpointPanel.add(panel);
+
```
Newline

deathbeam commented on runelite-client/src/main/java/net/runelite/client/plugins/hiscore/HiscorePanel.java:
```diff
@@ -306,172 +226,94 @@ public void mouseClicked(MouseEvent e)
+						{
+							return;
+						}
+						executor.execute(HiscorePanel.this::lookup);
```
Add newline above

deathbeam commented on runelite-client/src/main/java/net/runelite/client/plugins/hiscore/HiscorePanel.java:
```diff
@@ -602,33 +470,198 @@ else if (result.getSkill(skill) != null && result.getSkill(skill).getRank() != -
 
 				label.setText(Integer.toString(level));
 			}
+			label.setToolTipText(detailsHtml(skill));
```
Add newline above

deathbeam commented on runelite-client/src/main/java/net/runelite/client/plugins/hiscore/HiscorePanel.java:
```diff
@@ -602,33 +470,198 @@ else if (result.getSkill(skill) != null && result.getSkill(skill).getRank() != -
+			case 5:
+				return HiscoreSkill.BOUNTY_HUNTER_HUNTER;
+		}
+		return null;
```
Add newline above

deathbeam commented on runelite-client/src/main/java/net/runelite/client/plugins/info/InfoPanel.java:
```diff
@@ -52,6 +63,14 @@
+	private static ImageIcon ARROW_RIGHT_ICON;
+	private static ImageIcon GITHUB_ICON;
+	private static ImageIcon DISCORD_ICON;
+	private static ImageIcon PATREON_ICON;
```
Mark these 4 fields final too and move all constants together at top of the class

deathbeam commented on runelite-client/src/main/java/net/runelite/client/plugins/kourendlibrary/BookPanel.java:
```diff
@@ -36,6 +37,10 @@
 
 	BookPanel(Book b)
 	{
+
```
Newline

deathbeam commented on runelite-client/src/main/java/net/runelite/client/plugins/kourendlibrary/KourendLibraryPanel.java:
```diff
@@ -70,21 +86,65 @@ void init()
+				resetImg = ImageIO.read(KourendLibraryPanel.class.getResourceAsStream("reset.png"));
+				resetClickImg = SwingUtil.grayscaleOffset(resetImg, -100);
+			}
+			resetIcon = new ImageIcon(resetImg);
```
Newline above

psikoi commented on runelite-client/src/main/java/net/runelite/client/plugins/grandexchange/GrandExchangeSearchPanel.java:
```diff
@@ -26,46 +27,89 @@
+		{
+			synchronized (ImageIO.class)
+			{
+				SEARCH_ICON = new ImageIcon(ImageIO.read(GrandExchangeSearchPanel.class.getResourceAsStream("search.png")));
```
@deathbeam

psikoi commented on runelite-client/src/main/java/net/runelite/client/plugins/config/ConfigPanel.java:
```diff
@@ -301,7 +345,7 @@ public void onActivate()
 	{
 		removeAll();
-		add(new JLabel("Plugin Configuration", SwingConstants.CENTER));
+		add(new JLabel("Plugin Configuration", SwingConstants.LEFT));
```
Didn't catch that, change the label to just Configuration? why wouldn't it fit? it's smaller

psikoi commented on runelite-client/src/main/java/net/runelite/client/plugins/feed/FeedPlugin.java:
```diff
@@ -93,6 +93,7 @@ protected void startUp() throws Exception
 		navButton = NavigationButton.builder()
 			.name("News Feed")
 			.icon(icon)
+			.priority(8)
```
What is the alternative? while keeping this order

psikoi commented:
Following @deathbeam 's review, some changes were made to the code, mainly clean-ups but a few notable changes were:

- The config panel's title has now been renamed "Configuration" (previously "Plugin Configuration")

Some changes I also made this update were:

- Changed the text on Grand Exchange sell offers from "Spent: x/y" to "Received: x/y"
- Refactored the Grand Exchange offers panel into it's own panel (GrandExchangeOffersPanel) as it was previously all done in the GrandExchangePanel, and as development progressed, this file became larger for no reason.
- Changed the layout on the Grand Exchange Offers list as the previous one worked poorly with resizable mode
- Changed the red tone on the log out button (top toolbar) to a brighter one.

psikoi commented on runelite-client/src/main/java/net/runelite/client/util/StackFormatter.java:
```diff
@@ -48,7 +49,7 @@
 	 * A number formatter
 	 */
-	private static final NumberFormat NUMBER_FORMATTER = NumberFormat.getInstance();
+	private static final NumberFormat NUMBER_FORMATTER = NumberFormat.getInstance(Locale.UK);
```
Do I need to remove it from the unit testing?

psikoi commented:
Following deathbeam's review (again), a few small changes were made to the code, mostly clean-up.

- Moved search/loading/error icons used in the search bars, to a generic location under resources/ui/components and removed them from their individual plugins (ge/config/hiscores).
- Removed locale specification on the StackFormatter and specified Locale.ENGLISH for unit testing
- Incorporated the new auto complete update into my hiscores redesign commit
- Fixed a bug with the g.e offers that was preventing some slots from updating on the ui.

Ready for review and testing again.

PariksitDogra commented:
Is there anyway to pull this branch of RL? Would love some dark mode.

deathbeam commented on runelite-client/src/main/java/net/runelite/client/plugins/xptracker/XpPanel.java:
```diff
@@ -29,64 +30,93 @@
+	final JPanel overallPanel = new JPanel();
+
+	/* This displays the "No exp gained" text */
+	final PluginErrorPanel errorPanel = new PluginErrorPanel();
```
Make this private

psikoi commented:
A couple of graphical changes were made:

- Config panel now has a shorter height gap between config items.
- Config panel's title is now white and with consistent margins to other panels' titles.
- Farming panel now has consistent margins.
- Hiscores panel now has consistent margins.
- Kourend Library panel's title is now white and with consistent margins to other panels' titles.
- Notes panel's title is now white and with consistent margins to other panels' titles.
- News Feed panel's title is now white and with consistent margins to other panels' titles.
- The selected item effect on combo boxes is now more noticeable.

deathbeam commented:
Updated the PR to work with the new skilling calculator changes. Here is how it looks atm:

![screenie](https://user-images.githubusercontent.com/5115805/40044784-b618d712-5828-11e8-9185-4770f418677c.png)

psikoi commented:
This gets delayed every week and new features keep coming in, great. I'll work on the skill calcs today/tomorrow.

psikoi commented:
- Reworded the Exp tracker panels
- Added 10px padding to the notes editor
- Added new commit for the skilling calcs redesign, currently empty, will start work on this in a few hours.

We're back in business boooiiis

psikoi commented:
- Added quick-fix commit to the skill calcs plugin, will continue development on it after the merge.
